packages/legend-studio/src/models/protocols/pure/v1/transformation/pureGraph/to/V1_ProtocolToMetaModelPropertyMappingVisitor.ts

```diff
@@ -442,7 +448,17 @@ export class V1_ProtocolToMetaModelPropertyMappingVisitor
     );
     const relationalPropertyMapping = new RelationalPropertyMapping(
       this.topParent ?? this.immediateParent,
-      PropertyExplicitReference.create(property),
+      propertyOwner instanceof Class // TODO: we also probably need to handle this for association mapping
```

Comment:
Confused as to why the owner determines impl vs expli property type

Reply:
Doesn't it make sense to check if the immediate parent is RootRelational -> expl if not use impl?